ast/src/parsed/asm.rs

```diff
@@ -11,9 +11,15 @@ pub struct ASMFile<T> {
 pub struct Machine<T> {
     pub start: usize,
     pub name: String,
+    pub arguments: MachineArguments,
     pub statements: Vec<MachineStatement<T>>,
 }
 
+#[derive(Debug, PartialEq, Eq, Default)]
+pub struct MachineArguments {
+    pub latch: Option<String>,
```

Comment:
These are the arguments in the machine's definition, right? Like the latch and signature that you showed me earlier

Reply:
yes, only the latch actually as the arguments are on the functions